### PR TITLE
[ONNX] Export 'aten::resolve_neg' and 'aten::resolve_conj' as Noop in ONNX

### LINKED ISSUE #73619: ONNX export fails with a `resolve_conj` op when a tensor slice is printed.
### 🐛 Describe the bug

Exporting a module as ONNX fails when the module involves a tensor slice being (maybe also any view) passed to `print()`. The export fails with a message stating that the `resolve_conj` op is not available.

It appears that printing a view invokes `resolve_conj`, which causes ONNX exports to fail even in cases where the branch is extraneous. As far as I understand, the ONNX optimiser can remove these kinds of extraneous branches, but in this case it fails before pruning.

Example:
```python
import torch
from torch import nn

class BadFirst(nn.Module):
    def __init__(self):
        super().__init__()

    def forward(self, x):
        x_firsts = x[:, 0]
        print(f"x_firsts: {x_firsts}") # removing this line fixes the issue.
        return x_firsts


m = BadFirst().eval()
x = torch.rand(10, 5)

res = m(x) # this works
torch.onnx.export(m, x, "m.onnx") # this fails due to a resolve_conj op
```
Observed Output:
```
x_firsts: tensor([0.8206, 0.7061, 0.0366, 0.7012, 0.4380, 0.0920, 0.7127, 0.1474, 0.5061,
        0.9991])
x_firsts: tensor([0.8206, 0.7061, 0.0366, 0.7012, 0.4380, 0.0920, 0.7127, 0.1474, 0.5061,
        0.9991])
Traceback (most recent call last):
  File "~\scratch\pytorch\example.py", line 18, in <module>
    torch.onnx.export(m, x, "m.onnx") # this fails due to a resolve_conj op
  File "~\.virtualenvs\pytorch-jk_rFARN\lib\site-packages\torch\onnx\__init__.py", line 316, in export
    return utils.export(model, args, f, export_params, verbose, training,
  File "~\.virtualenvs\pytorch-jk_rFARN\lib\site-packages\torch\onnx\utils.py", line 107, in export
    _export(model, args, f, export_params, verbose, training, input_names, output_names,
  File "~\.virtualenvs\pytorch-jk_rFARN\lib\site-packages\torch\onnx\utils.py", line 724, in _export
    _model_to_graph(model, args, verbose, input_names,
  File "~\.virtualenvs\pytorch-jk_rFARN\lib\site-packages\torch\onnx\utils.py", line 497, in _model_to_graph
    graph = _optimize_graph(graph, operator_export_type,
  File "~\.virtualenvs\pytorch-jk_rFARN\lib\site-packages\torch\onnx\utils.py", line 216, in _optimize_graph
    graph = torch._C._jit_pass_onnx(graph, operator_export_type)
  File "~\.virtualenvs\pytorch-jk_rFARN\lib\site-packages\torch\onnx\__init__.py", line 373, in _run_symbolic_function
    return utils._run_symbolic_function(*args, **kwargs)
  File "~\.virtualenvs\pytorch-jk_rFARN\lib\site-packages\torch\onnx\utils.py", line 1028, in _run_symbolic_function
    symbolic_fn = _find_symbolic_in_registry(domain, op_name, opset_version, operator_export_type)
  File "~\.virtualenvs\pytorch-jk_rFARN\lib\site-packages\torch\onnx\utils.py", line 982, in _find_symbolic_in_registry
    return sym_registry.get_registered_op(op_name, domain, opset_version)
  File "~\.virtualenvs\pytorch-jk_rFARN\lib\site-packages\torch\onnx\symbolic_registry.py", line 125, in get_registered_op
    raise RuntimeError(msg)
RuntimeError: Exporting the operator resolve_conj to ONNX opset version 9 is not supported. Please feel free to request support or submit a pull request on PyTorch GitHub.
```
I think I understand why `resolve_conj` is invoked here, but causing an ONNX export failure feels like an unintentional side-effect, that could potentially be avoided by pruning the graph before invoking the ONNX optimiser.

If `torch.jit.script` is run first, then exporting the ONNX model succeeds, but `torch.jit.trace` fails in the same way.

### Versions

Test 1 (python 3.9.6/torch 1.10.2):
```
PyTorch version: 1.10.2+cpu
Is debug build: False
CUDA used to build PyTorch: Could not collect
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 10 Enterprise
GCC version: Could not collect
Clang version: Could not collect
CMake version: version 3.21.0-rc3
Libc version: N/A

Python version: 3.9.6 (tags/v3.9.6:db3ff76, Jun 28 2021, 15:26:21) [MSC v.1929 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.19042-SP0
Is CUDA available: False
CUDA runtime version: 10.0.130
GPU models and configuration: GPU 0: Quadro P400
Nvidia driver version: 461.55

cuDNN version: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0\bin\cudnn64_7.dll
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.22.2
[pip3] torch==1.10.2
[conda] Could not collect
```
Test 2 (python 3.9.10/torch 1.9.1):
```
PyTorch version: 1.9.1+cpu
Is debug build: False
CUDA used to build PyTorch: Could not collect
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 10 Enterprise
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.9.10 (tags/v3.9.10:f2f3f53, Jan 17 2022, 15:14:21) [MSC v.1929 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.19042-SP0
Is CUDA available: False
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: Quadro P2000
Nvidia driver version: 462.30
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.22.2
[pip3] torch==1.9.1
[conda] Could not collect
```

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86180

Fixes #73619

### CODE REVIEW

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -6302,6 +6302,18 @@ def is_pinned(g: jit_utils.GraphContext, self, device=None):
     return None
 
 
+@_onnx_symbolic("aten::resolve_conj")
+def resolve_conj(g: jit_utils.GraphContext, self: _C.Value):
```
Comment (justinchuby) [inline torch/onnx/symbolic_opset9.py:6306]:
```suggestion
def resolve_conj(g: jit_utils.GraphContext, self: _C.Value) -> _C.Value:
```

**Review by justinchuby — APPROVED** [review]

**Comment by justinchuby** [comment]
Should we handle when the inputs are complex and we actually want to take the conjugate? I don't remember if this is supported yet. If not maybe a TODO comment?

**Comment by BowenBao** [comment]
> Should we handle when the inputs are complex and we actually want to take the conjugate? I don't remember if this is supported yet. If not maybe a TODO comment?

These two ops are also noops for ONNX complex. It's not taking conjugate, but rather materializing it since torch stores additional conjugate bit. So it's only adjusting the internal representation but not its "value", and from ONNX perspective this op does nothing. Adding to the comment.

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -6302,6 +6302,18 @@ def is_pinned(g: jit_utils.GraphContext, self, device=None):
...
+    # Noop in ONNX.
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:6307]:
```suggestion
    # Noop in ONNX. This op is materializing conjugation w.r.t conjugate bit. 
    # It is only adjusting the internal representation in torch but not its "value".
    # Hence from ONNX perspective this op does nothing.
```

File: torch/onnx/symbolic_opset9.py [hunk 3]
```diff
@@ -6302,6 +6302,18 @@ def is_pinned(g: jit_utils.GraphContext, self, device=None):
...
+    return self
+
+
+@_onnx_symbolic("aten::resolve_neg")
+def resolve_neg(g: jit_utils.GraphContext, self: _C.Value):
+    # Noop in ONNX.
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:6313]:
```suggestion
    # Noop in ONNX. This op is materializing negation w.r.t negative bit. 
    # It is only adjusting the internal representation in torch but not its "value".
    # Hence from ONNX perspective this op does nothing.
```

File: torch/onnx/symbolic_opset9.py [hunk 4]
```diff
@@ -6302,6 +6302,18 @@ def is_pinned(g: jit_utils.GraphContext, self, device=None):
...
 
 
+@_onnx_symbolic("aten::resolve_conj")
+def resolve_conj(g: jit_utils.GraphContext, self: _C.Value):
+    # Noop in ONNX.
+    return self
+
+
+@_onnx_symbolic("aten::resolve_neg")
+def resolve_neg(g: jit_utils.GraphContext, self: _C.Value):
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:6312]:
```suggestion
def resolve_neg(g: jit_utils.GraphContext, self: _C.Value) -> _C.Value:
```

File: torch/onnx/symbolic_opset9.py [hunk 5]
```diff
@@ -6302,6 +6302,22 @@ def is_pinned(g: jit_utils.GraphContext, self, device=None):
     return None
 
 
+@_onnx_symbolic("aten::resolve_conj")
+def resolve_conj(g: jit_utils.GraphContext, self: _C.Value) -> _C.Value:
+    # Noop in ONNX. This op is materializing conjugation w.r.t conjugate bit.
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:6307]:
```suggestion
    # Noop in ONNX. This op is materializing conjugation w.r.t conjugate bit.
```

File: torch/onnx/symbolic_opset9.py [hunk 6]
```diff
@@ -6302,6 +6302,22 @@ def is_pinned(g: jit_utils.GraphContext, self, device=None):
     return None
 
 
+@_onnx_symbolic("aten::resolve_conj")
+def resolve_conj(g: jit_utils.GraphContext, self: _C.Value) -> _C.Value:
```
Comment (justinchuby) [inline torch/onnx/symbolic_opset9.py:6306]: Need to add these to `__all__`

Comment (thiagocrepaldi) [inline torch/onnx/symbolic_opset9.py:6306]: `def _conj` is needed too. By setting `GLABLS.onnx_shape_inference=False`, the exporter would advance a little further and complain about the missing `aten::_conj`. See https://github.com/pytorch/pytorch/pull/86223 which has all 3 symbolics, plus some type checking to make sure input is never complex types

File: torch/onnx/symbolic_opset9.py [hunk 7]
```diff
@@ -6302,6 +6302,22 @@ def is_pinned(g: jit_utils.GraphContext, self, device=None):
...
+    # Noop in ONNX. This op is materializing conjugation w.r.t conjugate bit.
+    # It is only adjusting the internal representation in torch but not its "value".
+    # Hence from ONNX perspective this op does nothing.
+    return self
+
+
+@_onnx_symbolic("aten::resolve_neg")
+def resolve_neg(g: jit_utils.GraphContext, self: _C.Value) -> _C.Value:
```
Comment (AllenTiTaiWang) [inline torch/onnx/symbolic_opset9.py:6314]: @justinchuby Do we need beartype in this function?

**Review by titaiwangms — APPROVED** [review]

File: torch/onnx/symbolic_opset9.py (hunk 7 above)
Comment (justinchuby) [reply torch/onnx/symbolic_opset9.py:6314]: We do actually. Good point

File: torch/onnx/symbolic_opset9.py (hunk 6 above)
Comment (BowenBao) [reply torch/onnx/symbolic_opset9.py:6306]: Cool, I'll close this one and focus on your version. Will be pasting discussion and comments there.

**Comment by BowenBao** [comment]
Closing as duplicate of #86223

**Pull request closed without merging**